src/overflow.ts

```diff
@@ -0,0 +1,73 @@
+import { AttributePart, noChange } from 'lit-html';
+import { Directive, directive, DirectiveResult } from 'lit-html/directive.js';
+
+type OnOverflow = (
+	visibleElements: Set<HTMLElement>,
+	overflownElements: Set<HTMLElement>,
+) => void;
+
+function isEntryHidden(el: IntersectionObserverEntry) {
+	return el.boundingClientRect.height === 0;
+}
+
+class OverflowDirective extends Directive {
+	_observer?: IntersectionObserver;
+	_container?: HTMLElement | null;
+
+	render() {
+		return noChange;
+	}
+
+	update(part: AttributePart, [onOverflow]: [OnOverflow]) {
+		const slot = part.element as HTMLSlotElement;
```

Comment:
You should assert that it is a slot element by checking the tagName and throwing an error.